scripts/validate_string_concatenation.py

```diff
@@ -1,51 +1,313 @@
 #!/usr/bin/env python
 """
-GH #30454
+Unwanted patterns test cases.
 
-Check where there is a string that needs to be concatenated.
+The reason this file exist despite the fact we already have
+`ci/code_checks.sh`,
+(see https://github.com/pandas-dev/pandas/blob/master/ci/code_checks.sh)
 
-This is necessary after black formating,
-where for example black transforms this:
-
->>> foo = (
-...         "bar "
-...         "baz"
-...     )
-
-into this:
-
->>> foo = ("bar " "baz")
-
-Black is not considering this as an
-issue (see issue https://github.com/psf/black/issues/1051),
-so we are checking it here.
+is that some of the test cases are more complex/imposible to validate via regex.
+So this file is somewhat an extensions to `ci/code_checks.sh`
 """
 
 import argparse
 import os
 import sys
 import token
 import tokenize
-from typing import Generator, List, Tuple
+from typing import Callable, Generator, List, Tuple
 
 FILE_EXTENSIONS_TO_CHECK = (".py", ".pyx", ".pyx.ini", ".pxd")
 
 
-def main(source_path: str, output_format: str) -> bool:
+def bare_pytest_raises(source_path: str) -> Generator[Tuple[str, int, str], None, None]:
+    """
+    Test Case for bare pytest raises.
+
+    For example, this is wrong:
+
+    >>> with pytest.raise(ValueError):
+    ...     # Some code that raises ValueError
+
+    And this is what we want instead:
+
+    >>> with pytest.raise(ValueError, match="foo"):
+    ...     # Some code that raises ValueError
+
+    Parameters
+    ----------
+    source_path : str
+        File path pointing to a single file.
+
+    Yields
+    ------
+    source_path : str
+        Source file path.
+    line_number : int
+        Line number of unconcatenated string.
+    msg : str
+        Explenation of the error.
+
+    Notes
+    -----
+    GH #23922
+    """
+    with open(source_path, "r") as file_name:
+        tokens: List = list(tokenize.generate_tokens(file_name.readline))
+
+    for counter, current_token in enumerate(tokens, start=1):
+        if not (current_token.type == token.NAME and current_token.string == "raises"):
+            continue
+        for next_token in tokens[counter:]:
+            if next_token.type == token.NAME and next_token.string == "match":
+                break
+            # token.NEWLINE refers to the end of a logical line
+            # unlike token.NL or "\n" which represents a newline
+            if next_token.type == token.NEWLINE:
+                yield (
+                    source_path,
+                    current_token.start[0],
+                    "Bare pytests raise have been found.",
```

Comment:
Probably worth adding what to do, I think this message can be a bit cryptic for beginners.